packages/auth/src/api/index.ts

```diff
@@ -265,11 +269,21 @@ export function _getFinalTarget(
 ): string {
   const base = `${host}${path}?${query}`;
 
-  if (!(auth as AuthInternal).config.emulator) {
-    return `${auth.config.apiScheme}://${base}`;
+  const finalTarget = (auth as AuthInternal).config.emulator
+    ? _emulatorUrl(auth.config as ConfigInternal, base)
+    : `${auth.config.apiScheme}://${base}`;
+
+  // TODO get the exchange URL from the persistence method
+  //      don't use startsWith v1/accounts...
+  if (
+    (auth as AuthInternal)._getPersistence() === PersistenceType.COOKIE &&
+    (path.startsWith('/v1/accounts:signIn') || path === Endpoint.TOKEN)
+  ) {
+    const params = new URLSearchParams({ finalTarget });
+    return `${window.location.origin}/__cookies__?${params.toString()}`;
```

Comment:
I think this function can be called in Node so we may have to make sure `window` doesn't throw. I'm not sure if the `=== PersistenceType.COOKIE` is a guarantee that we're in browser, maybe safer to explicitly check for window. What's the "don't use startsWith v1/accounts" comment about?

Reply:
SG. My thought with "don't use startsWith v1/accounts" is to test for enum inclusion rather than the string, e.g, `SignInEndpoints.includes(path)`